packages/core/src/render3/instructions/shared.ts

```diff
@@ -1385,16 +1386,17 @@ function findDirectiveDefMatches(
         diPublicInInjector(getOrCreateNodeInjectorForNode(tNode, viewData), tView, def.type);
 
         if (isComponentDef(def)) {
-          ngDevMode &&
-              assertNodeOfPossibleTypes(
-                  tNode, [TNodeType.Element],
-                  `"${tNode.tagName}" tags cannot be used as component hosts. ` +
-                      `Please use a different tag to activate the ${
-                          stringify(def.type)} component.`);
-          if (tNode.flags & TNodeFlags.isComponentHost) throwMultipleComponentError(tNode);
-          markAsComponentHost(tView, tNode);
-          // The component is always stored first with directives after.
-          matches.unshift(def);
+          if (ngDevMode) {
+            assertNodeOfPossibleTypes(
+                tNode, [TNodeType.Element],
+                `"${tNode.tagName}" tags cannot be used as component hosts. ` +
+                    `Please use a different tag to activate the ${stringify(def.type)} component.`);
+
+            if (tNode.flags & TNodeFlags.isComponentHost) throwMultipleComponentError(tNode);
+            markAsComponentHost(tView, tNode);
+            // The component is always stored first with directives after.
+            matches.unshift(def);
+          }
```

Comment:
I think the ngDevMode condition should end at `throwMultipleComponentError` call:

```suggestion
          if (ngDevMode) {
            assertNodeOfPossibleTypes(
                tNode, [TNodeType.Element],
                `"${tNode.tagName}" tags cannot be used as component hosts. ` +
                    `Please use a different tag to activate the ${stringify(def.type)} component.`);

            if (tNode.flags & TNodeFlags.isComponentHost) throwMultipleComponentError(tNode);
          }
          markAsComponentHost(tView, tNode);
          // The component is always stored first with directives after.
          matches.unshift(def);
```

Reply:
Good catch @AndrewKushnir - I did not see this because of the number of curly braces/indentations this code has 😄